Swiftification/String.swift

```diff
@@ -53,6 +53,15 @@ public extension String {
         return result
     }
 
+    /// Returns a value with string inserted at index.
+    @warn_unused_result
+    func insert(string: String?, atIndex index: Int) -> String {
```

Comment:
For example, the corresponding function on Array is http://swiftdoc.org/v3.0/type/Array/#func-insert_at_. Incidentally, that function is mutating. I don't see an immutable version like this one, so I'd be inclined to follow the same pattern here.